pkg/kubectl/cmd/create_job.go

```diff
@@ -118,29 +152,28 @@ func (c *CreateJobOptions) RunCreateJob() error {
 	if !ok {
 		return fmt.Errorf("from must be an existing cronjob")
 	}
-
-	return c.createJob(cronJob)
-}
-
-func (c *CreateJobOptions) createJob(cronJob *batchv1beta1.CronJob) error {
-	annotations := make(map[string]string)
-	annotations["cronjob.kubernetes.io/instantiate"] = "manual"
-	for k, v := range cronJob.Spec.JobTemplate.Annotations {
-		annotations[k] = v
-	}
-	jobToCreate := &batchv1.Job{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        c.Name,
-			Namespace:   c.Namespace,
-			Annotations: annotations,
-			Labels:      cronJob.Spec.JobTemplate.Labels,
-		},
-		Spec: cronJob.Spec.JobTemplate.Spec,
+	generator := &kubectl.JobGeneratorV1{
+		Name: c.Name,
+		From: &cronJob.Spec.JobTemplate,
```

Comment:
No.  There's no reason to complicate this with a generator for indirection.  You have the JobTemplate, just create it.